src/table.c

```diff
@@ -0,0 +1,135 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include "table.h"
+
+/* Função para criar e inicializar uma nova tabela hash, com n
+ * linhas (n = módulo da função hash).
+ * Retorna a tabela ou NULL em caso de erro.
+ */
+struct table_t *table_create(int n){
+    struct table_t *table;
+    table = (struct table_t*) calloc(1,sizeof(struct table_t));
+    table->size = n;
+    table->lists = (struct list_t**) calloc(n,sizeof(struct list_t*)); 
+    return table;
+}
+
+/* Função que elimina uma tabela, libertando *toda* a memória utilizada
+ * pela tabela.
+ * Retorna 0 (OK) ou -1 em caso de erro.
+ */
+int table_destroy(struct table_t *table){
+    int status = 0;
+    for(int i=0; i<table->size; i++){
+        status = list_destroy(table->lists[i]);
+        if (status != 0) return status;
+    }
+    return status;
+
+}
+
+/* Função que calcula o índice da lista a partir da chave
+ */
+int hash_code(char *key, int n){
+    //does something, e.g.
+    int hashed = strlen(key);
```

Comment:
Yeah like you said, this is good for testing purposes, but a more complex hash code would probably be better and we can work on it later :)

Reply:
fuck this was meant to be a single comment